Source/Model/FeatureConfig/AppLock/AppLockController.swift

```diff
@@ -133,7 +133,10 @@ public final class AppLockController: AppLockType {
         
         let canEvaluatePolicy = context.canEvaluatePolicy(scenario.policy, error: &error)
         
-        if scenario.supportsUserFallback && (BiometricsState.biometricsChanged(in: context) || !canEvaluatePolicy) {
+        let biometricsChanged = BiometricsState.biometricsChanged(in: context) && config.useBiometricsOrCustomPasscode
```

Comment:
Isn't `useBiometricsOrCustomPasscode` already part of the `AuthenticationScenario`? Seems like `AuthenticationScenario.supportsUserFallback` could take this into account. 

Then we could simplify this to:

```suggestion
        let biometricsChanged = BiometricsState.biometricsChanged(in: context)
```

Reply:
I think there may be some confusion (on my part) about what exactly `supportsUserFalback` is supposed to mean. I took it to understand that there is a secondary authentication method available. But is it suppose to mean that the fallback must be the custom passcode?

It's confusing because if you have biometrics enabled, you must also have a device pin set (which is also used as a fallback).

I'm thinking now that Jacob's suggest would work with:

```swift
var supportsUserFallback: Bool {
  switch self {
    case let .screenLock(requireBiometrics):
      return requireBiometrics
    case .databaseLock:
      return false
  }
}
```

Reply:
Probably also we should rename `supportsUserFallback` to something like `shouldFallbackToCustomPasscode`.

Reply:
>It's confusing because if you have biometrics enabled, you must also have a device pin set (which is also used as a fallback).

But we don't explicitly request the PIN code as fallback do we? It happen automatically by the OS. When EAR is enabled `supportsUserFallback` is always false but you can't still unlock your phone with the PIN if the biometric fails.

Reply:
Yes it's handled by the OS, but the naming `supportsUserFallback` doesn't make that so clear (leading to my confusion).

Reply:
> When EAR is enabled `supportsUserFallback` is always false but you can't still unlock your phone with the PIN if the biometric fails.

Do you mean "... but you **can** still unlock your phone with the PIN""?

If `supportsUserFallback` is false, then it means that we shouldn't allow the user to enter via a custom passcode? I'm wondering what it means if EAR is enabled and `useBiometricsOrCustomPasscode` is `true`. I guess it means that you must use biometrics only and nothing else.

Reply:
`useBiometricsOrCustomPasscode` doesn't affect EAR. 
```
var policy: LAPolicy {
            switch self {
            case .screenLock(requireBiometrics: let requireBiometrics):
                return requireBiometrics ? .deviceOwnerAuthenticationWithBiometrics : .deviceOwnerAuthentication
            case .databaseLock:
                return .deviceOwnerAuthentication
            }
        }
```

Reply:
@typfel @johnxnguyen I replaced `var supportsUserFallback: Bool ` with `var supportedUserFallback: UserFallback`
And 
```
enum UserFallback {
        /// User is allowed to use only a custom passcode as a fallback
        case customPasscode
        /// User is allowed to use only a device passcode as a fallback
        case devicePasscode
        /// User is allowed to use any fallback
        case all
    }
```
Because we have different use cases for `UserFallback`.